components/utils/hash-generator.utils.ts

```diff
@@ -0,0 +1,24 @@
+import crypto, { BinaryToTextEncoding } from "crypto";
+
+export const generateHash = (
+  algorithm: string,
+  data: string,
+  encoding: BinaryToTextEncoding = "hex",
+  secretKey?: string
+): string => {
+  if (typeof data !== "string" || data.trim() === "") {
+    throw new Error("Data must be a non-empty string");
+  }
+
+  try {
+    const hash = algorithm.startsWith("hmac")
+      ? crypto.createHmac(algorithm.replace("hmac-", ""), secretKey!)
```

Comment:
Let's write a comment why we used `!` for `secretKey`
I saw later that it's optional in UI, but I was confused while doing the review.

Reply:
Oh, I see now, this is a bit misleading.

Initially we defined `secretKey` as optional, but then used `!` operator, implying it's always available.
I checked the code, it will work since we always have a fallback:

```
const resolvedSecretKey = secretKey || crypto.randomBytes(32).toString("hex");
```
While this works it's not ideal and it might confuse developers in future. 
It can be misleading and potentially lead to bugs.
Safer approach to this would be to explicitly check for `secretKey`

```
if (!secretKey) {
  throw new Error("Secret key must be provided for HMAC algorithms");
}
```

If you would still like to keep the same pattern then it should happen in this function, something like this:

```
const secret = secretKey || crypto.randomBytes(32).toString("hex");
```

Reply:
I've implemented both validations as suggested. For the check:

```
if (!secretKey) {
  throw new Error("Secret key must be provided for HMAC algorithms");
}
```
I added `algorithm.startsWith('hmac')` validation, since the secret key is only required for HMAC algorithms.

Thanks for the valuable insight!